pkg/operator/startupmonitorreadiness/readiness_checks_test.go

```diff
@@ -330,6 +329,16 @@ func TestGoodReadyzEndpoint(t *testing.T) {
 			msg:       "waiting for kube-apiserver static pod to listen on port 6443",
 			customURL: "https://localhost:1234",
 		},
+		{
+			name: "scenario 6: unexpected err from the server",
+			rspWriterFn: func(w http.ResponseWriter) {
+				panic("bum")
```

Comment:
mhm, won't this fail the test ?
I haven't found any recover routine. 
Why not to:
```
	rspWriterFn: func(w http.ResponseWriter) {
				w.WriteHeader(http.StatusInternalServerError)
				w.Write([]byte("kube is on fire"))
			},
``` 
?

Reply:
iiuc `setupServerClient` will panic, but the test continues and `msg` is still being checked:
```
=== RUN   TestGoodHealthzEndpoint/scenario_3:_unexpected_err_from_the_server
2023/05/11 16:17:05 http: panic serving 127.0.0.1:40822: bum
goroutine 216 [running]:
net/http.(*conn).serve.func1()
        /usr/lib/sdk/golang/src/net/http/server.go:1854 +0xbf
panic({0x1dec240, 0x2507d80})
        /usr/lib/sdk/golang/src/runtime/panic.go:890 +0x263
github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/startupmonitorreadiness.TestGoodHealthzEndpoint.func2({0xc0000a2aa8?, 0x4ac693?})
        /var/home/vrutkovs/src/github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/startupmonitorreadiness/readiness_checks_test.go:412 +0x27
github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/startupmonitorreadiness.TestGoodHealthzEndpoint.func4.2({0x252d8e0, 0xc00077c540}, 0xc000722a00)
        /var/home/vrutkovs/src/github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/startupmonitorreadiness/readiness_checks_test.go:455 +0x1b4
github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/startupmonitorreadiness.setupServerClient.func1({0x252d8e0?, 0xc00077c540?}, 0x1ef57a0?)
        /var/home/vrutkovs/src/github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/startupmonitorreadiness/readiness_checks_test.go:583 +0x27
net/http.HandlerFunc.ServeHTTP(0xc00073a5fd?, {0x252d8e0?, 0xc00077c540?}, 0x47048e?)
        /usr/lib/sdk/golang/src/net/http/server.go:2122 +0x2f
net/http.serverHandler.ServeHTTP({0xc00045dc80?}, {0x252d8e0, 0xc00077c540}, 0xc000722a00)
        /usr/lib/sdk/golang/src/net/http/server.go:2936 +0x316
net/http.(*conn).serve(0xc000537710, {0x252e800, 0xc00045db60})
        /usr/lib/sdk/golang/src/net/http/server.go:1995 +0x612
created by net/http.(*Server).Serve
        /usr/lib/sdk/golang/src/net/http/server.go:3089 +0x5ed
    /var/home/vrutkovs/src/github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/startupmonitorreadiness/readiness_checks_test.go:573: unexpected message "failed sending request to kube-apiserver: Get \"http://127.0.0.1:41369/healthz?verbose=true&exclude=api-openshift-apiserver-available\": EOF", expected "/healthz?verbose=true&exclude=-openshift-apiserver-available\": EOF"
```
is what I get when a wrong message is expected.

I copied this test case from `readyz` to have an additional check for request parameters

Reply:
mhm, if nothing recovers from a panic does it mean that not ALL tests are run :( ?!

compare: https://go.dev/play/p/4dnPJLOClis vs https://go.dev/play/p/LYaqICe9Z1P

Reply:
ping, I think we should fix these 2 tests.

Reply:
Right, let me figure out what's happening here

Reply:
Ah, there's `recover` in `net/http`: https://github.com/golang/go/blob/master/src/net/http/server.go#L1863